pandas/core/dtypes/dtypes.py

```diff
@@ -564,6 +622,17 @@ def construct_from_string(cls, string):
                 pass
         raise TypeError("could not construct PeriodDtype")
 
+    @classmethod
+    def construct_from_string_strict(cls, string):
+        """
+        Strict construction from a string, raise a TypeError if not
+        possible
+        """
+        if string.startswith('period[') or string.startswith('Period['):
+            # do not parse string like U as period[U]
+            return PeriodDtype.construct_from_string(string)
+        raise TypeError("could not construct PeriodDtype")
```

Comment:
Should we consider rather changing `construct_from_string` itself? As those don't really adhere to the ExtensionDtype requirements I think?

If we keep both, I would make this one private.

Reply:
done